lib/trace.js

```diff
@@ -210,6 +212,18 @@ const _resolveDep = async ({
   bailOnMissing,
   addRootPackagePaths = false
 }) => {
+  // Support advanced Node.js ESM import types by ignoring built-in's as not
+  // needing any further inspection / inclusion.
+  if (IGNORE_IMPORT_PREFIXES.find((prefix) => depName.startsWith(prefix))) {
+    return null;
+  }
+
+  // Check for modern import `?` query imports and mutate dep name to remove.
+  const queryStringPosition = depName.indexOf("?");
+  if (queryStringPosition > -1) {
+    depName = depName.substring(0, queryStringPosition);
```

Comment:
Since we use `depName` elsewhere in this function, it seemed safer to just mutate the variable at the top before anything else started using it...